### PR TITLE
[FLASK] Add Snaps privacy warning on snap install

### PR DESCRIPTION
**Pull request by david0xd**
## Explanation
Fixes: https://github.com/MetaMask/MetaMask-planning/issues/406

This PR will add a way to display popup with privacy policy information when installing a snap. Popup will be displayed only once when first snap is installed and never again once agreed.

Implementation is done according to the UX/UI described here: https://www.figma.com/file/diidnCHx9FXj5LDiA5DZZr/Snaps-features?type=design&node-id=1150-49137&t=z6vvUV0ZgOT98ERJ-0

### Before
Screenshot 2023-05-08 at 16 38 20

### After
Screenshot 2023-05-09 at 12 20 57
Screenshot 2023-05-09 at 12 21 16

## Manual Testing Steps
Scenario A:
1. Remove any previous installation of MetaMask Flask.
2. Install MetaMask Flask.
3. Install arbitrary snap.
4. Popup with notice should be displayed.
5. Click 'OK' button to agree.

Scenario B:
1. Install another arbitrary snap after first one is installed and popup is displayed.
2. Popup should not be displayed anymore whenever new snap is installed.
3. Restart browser and try again or reload the extension.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by bowensanders** [comment]
Manual QA passes ✅

File: ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                {t('snapsPrivacyWarningSecondMessage')}
+              </Text>
+              <Text
+                variant={TextVariant.bodyMd}
+                className="snap-privacy-warning__more-details"
+              >
+                {t('click')}
+                <ButtonLink
+                  className="snap-privacy-warning__more-details__link"
+                  padding={0}
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.js:76]: Using size inherit should allow you to remove the styles added for this className and padding prop
```suggestion
                  className="snap-privacy-warning__more-details__link"
                  size={BUTTON_LINK_SIZES.INHERIT}
```

File: ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.stories.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+import SnapPrivacyWarning from '.';
+
+export default {
+  title: 'Components/App/snaps/SnapPrivacyWarning',
+
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.stories.js:7]: nit: remove line break
```suggestion
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Left some small suggestions

File: ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.js (hunk 1 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.js]: Ok, good, updated.

File: ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.stories.js (hunk 2 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.stories.js]: Updated.

**Comment by eriknson** [comment]
@georgewrmarshall unrelated to this specific PR, but feels like we should reduce / darken the backdrop opacity a bit to emphasis the modal more?

**Comment by bowensanders** [comment]
Manual QA passes again ✅

**Review by hmalik88 — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**